### ISSUE TITLE
Helm v 2.9.1 Error: Can't get a valid version for repositories even when the chart is available

### ISSUE THREAD

**Issue by dops-vijayv** [helm/helm#6543]
We're facing the below error during umbrella packaging of our product chart.
We're using a helm-docker image to pack helm charts like below.

> docker run --rm --user jenkins -v /home/jenkins/workspace/helm/pro:/pro:rw 
> --net=host --entrypoint=bash docker-registry-url.com/helm:v2.9.1-01 -c 
> 'helm repo add inprogress https://helm-repo-url/helm-inprogress/ && 
> helm repo add candidates https://helm-repo-url/helm-candidates/ && 
> helm package -u --save=0 -d pro /pro'
>  "inprogress" has been added to your repositories
>  "candidates" has been added to your repositories
>  Hang tight while we grab the latest from your chart repositories...
>  ...Unable to get an update from the "local" chart repository (http://127.0.0.1:8879/charts):
>  	Get http://127.0.0.1:8879/charts/index.yaml: dial tcp 127.0.0.1:8879: connect: connection refused
>  ...Successfully got an update from the "inprogress" chart repository
>  ...Successfully got an update from the "candidates" chart repository
> ...Successfully got an update from the "stable" chart repository
> ...Successfully got an update from the "incubator" chart repository
> Update Complete. ⎈Happy Helming!⎈
> Error: Can't get a valid version for repositories **chart01**. Try changing the version constraint in requirements.yaml
> script returned exit code 1

The chart01 has already been uploaded to the inprogress repo. The issue occurs for all dependent charts in a random fashion for no reason.

We're not sure if this issue is related to artifactory or helm itself.
Thanks!

**Comment by thomastaylor312** [comment]
What does the `Chart.yaml` look like for "chart01" and what does the entry in `requirements.yaml` look like? That should help me to answer the question better

**Comment by vijay-v6** [comment]
The issue is not with just chart01. It is with almost all charts that we pack. But it throws the error for different charts each time. When we checked the index.yaml in the helm-repo that we upload, it appears that the metadata of the specific chart is not being updated due to which it resulted in an error as above. What could be the reasons for the metadata about a chart not being updated in index.yaml during the build process?

**Comment by thomastaylor312** [comment]
I am not sure what could be causing it. I don't remember any bugs that sounded like this one, but is there anyway you can try with a more recent version of Helm? That could help rule out if it is a bug on the helm side or not

**Comment by dudubis** [comment]
Having the same issue. Never happened before, now we are stuck with different charts that can't be fetched even though they exist. What's happening?

**Comment by bacongobbler** [comment]
@dudubis can you please elaborate more on the issue you're seeing as well as how it appears, or how we can reproduce the failure locally?

**Comment by dudubis** [comment]
@bacongobbler index.yaml wasn't updated even though the chart was uploaded successfully. It never happened before and maybe a one time glitch, I will keep updating if I see it again.

**Comment by vijay-v6** [comment]
@thomastaylor312  @bacongobbler @dudubis 

I should've elaborated more on this issue.
We've multiple jenkins builds that runs in parallel. Some with difference of few seconds.
When 2 or 3 jobs gets to the helm chart publishing stage in jenkins, the chart info is not updated properly in the repo's index.yaml. 1 or 2 off the 3 builds, doesn't have it's relevant chart info in index.yaml even when it is published to the repo. We later found that the issue is with JFrog artifactory's helm repo (which we're using) and've logged a support request at their end. Details on it can be found here. [JFrog:RTFACT-20595](https://www.jfrog.com/jira/browse/RTFACT-20595)

Current workaround we've is that we reduced the number of concurrent builds in Jenkins from 10 to 3. But still the issue appears. So we've added a stage to reindex the helm repos before umbrella chart packaging. It still fails at times. But the frequency of failure because of this error has been reduced.

The issue can be closed as I feel there is nothing that could be done from helm's side.

Thanks!
